pkg/provisioner/create.go

```diff
@@ -23,7 +24,7 @@ func (p *Provisioner) CreateWorkspace(workspace *workspace.Workspace, target *pr
 	return err
 }
 
-func (p *Provisioner) CreateProject(project *workspace.Project, target *provider.ProviderTarget, cr *containerregistry.ContainerRegistry) error {
+func (p *Provisioner) CreateProject(project *workspace.Project, target *provider.ProviderTarget, cr *containerregistry.ContainerRegistry, gc *gitprovider.GitProviderConfig) error {
```

Comment:
`gc` is unused here. From what I can see, this is the only param that's making the `gitProvidersService` a dependency for the `Provisioner`. If the param is unused, we can remove the dependency as well.